src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs

```diff
@@ -136,7 +137,12 @@ internal static class BinderHelper
                 return new DynamicMetaObject(
                     Expression.Throw(
                         Expression.New(
-                            typeof(RuntimeBinderException).GetConstructor(new Type[] { typeof(string) }),
+                            typeof(RuntimeBinderException).GetConstructor(
+                                BindingFlags.NonPublic | BindingFlags.Instance,
```

Comment:
I think we may not have access to internal members via reflection in some situations. 
That is the reason why Linq.Expressions has a bunch of public XxxxOps helpers - they are public to be certainly callable via reflection.